composer/loggers/tqdm_logger.py

```diff
@@ -101,73 +102,44 @@ def _will_log(self, state: State, log_level: LogLevel) -> bool:
 
     def _log_metric(self, epoch: int, step: int, log_level: LogLevel, data: TLogData) -> None:
         del epoch, step, log_level  # Unused
-        pbar = self.pbar_val if self.is_validating else self.pbar_train
-        if pbar is None:
+        if self.is_train in self.pbars:
             # Logging outside an epoch
-            return
-        pbar.log_metric(data)
-
-    def _training_start(self, state: State, logger: Logger) -> None:
-        del state, logger  # Unused
-        if self.config is not None:
-            print("Config")
-            print("-" * 30)
-            yaml.safe_dump(self.config, stream=sys.stdout)
-            print("-" * 30)
-            print()
-
-    def epoch_start(self, state: State, logger: Logger) -> None:
-        del logger  # Unused
-        assert self.pbar_train is None
-        self.pbar_train = _TQDMLoggerInstance(total=state.steps_per_epoch,
-                                              epoch=state.epoch,
-                                              val=False,
-                                              keys_to_log=["loss/train"])
-
-    def after_backward(self, state: State, logger: Logger) -> None:
-        del state, logger  # Unused
-        assert self.pbar_train is not None
-        self.pbar_train.update()
-
-    def epoch_end(self, state: State, logger: Logger) -> None:
-        del state, logger  # Unused
-        assert self.pbar_train is not None
-        self.pbar_train.close()
-        self.pbar_train = None
-
-    def eval_start(self, state: State, logger: Logger) -> None:
-        del logger  # Unused
-        assert self.pbar_val is None
-        assert state.eval_dataloader is not None
-        self.pbar_val = _TQDMLoggerInstance(total=len(state.eval_dataloader),
-                                            epoch=state.epoch,
-                                            val=True,
-                                            keys_to_log=["accuracy/val"])
-        self.is_validating = True
-
-    def eval_after_forward(self, state: State, logger: Logger) -> None:
-        del state, logger  # Unused
-        assert self.pbar_val is not None
-        self.pbar_val.update()
-
-    def eval_end(self, state: State, logger: Logger) -> None:
-        del state, logger  # Unused
-        assert self.pbar_val is not None
-        self.pbar_val.close()
-        self.pbar_val = None
-        self.is_validating = False
+            assert self.is_train is not None
+            self.pbars[self.is_train].log_metric(data)
+
+    def _run_event(self, event: Event, state: State, logger: Logger) -> None:
```

Comment:
For readability, let's keep this logger implemented in the method-based approach. Can use common functions to keep this DRY.